src/framework/script/script-registry.js

```diff
@@ -36,6 +36,24 @@ class ScriptRegistry extends EventHandler {
         this.off();
     }
 
+    addSchema(script, schema) {
+
+        if (!schema) return;
+
+        if (!this.scriptSchemas) {
+            this.scriptSchemas = new Map();
+        }
+        this.scriptSchemas.set(script, schema);
+    }
+
+    getSchema(script) {
+        if (!this.scriptSchemas) {
+            return null;
+        }
+        const schema = this.scriptSchemas.get(script);
+        return schema && schema[script];
```

Comment:
is this part `schema[script]` needed here? isn't the line above already doing lookup by the `script`?
this whole function could perhaps be just
```
return this.scriptSchemas?.get(script);
```

Reply:
yep, valid